openlibrary/core/bookshelves_events.py

```diff
@@ -0,0 +1,115 @@
+from datetime import date, datetime
+from enum import IntEnum
+from . import db
+
+
+class BookshelfEvent(IntEnum):
+    START = 1
+    UPDATE = 2
+    FINISH = 3
+
+    @classmethod
+    def has_key(cls, key: str) -> bool:
```

Comment:
Will this work?  `has_key()` is kinda passe.
```suggestion
    def __contains__(cls, key: str) -> bool:
```
https://portingguide.readthedocs.io/en/latest/dicts.html

Reply:
Thanks for this note!  I tried this and was seeing `TypeDef` errors when checking if an int was in `BookshelfEvents`.  I applied this review note and fixed the error in #7212